src/Backups/BackupIO_Disk.cpp

```diff
@@ -77,4 +83,29 @@ void BackupWriterDisk::removeFiles(const Strings & file_names)
         disk->removeDirectory(path);
 }
 
+DataSourceDescription BackupWriterDisk::getDataSourceDescription() const
+{
+    return disk->getDataSourceDescription();
+}
+
+DataSourceDescription BackupReaderDisk::getDataSourceDescription() const
+{
+    return disk->getDataSourceDescription();
+}
+
+bool BackupWriterDisk::supportNativeCopy(DataSourceDescription data_source_description) const
+{
+    return data_source_description == disk->getDataSourceDescription();
+}
+
+void BackupWriterDisk::copyFileNative(std::shared_ptr<IDisk> from_disk, const String & file_name_from, const String & file_name_to)
+{
+    auto file_path = path / file_name_to;
+    disk->createDirectories(file_path.parent_path());
```

Comment:
Is `file_name_to` indeed a file name or may be it actually should be `file_path_to`? Because if judging by variable name it is just a file name and then here `file_path.parent_path() == path` and you always do in `disk->createDirectories(path);`, but `path` was passed in constructor of `BackupWriterDisk`, so kind of strange to always create the same directory in each method call.

Reply:
AFAIU it can be some relative prefix, but it doesn't look confusing to me? Also everywhere in BackupIO we use `file_name`.